frontend/src/Patients/PatientProfileModal.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import './PatientProfileModal.css';
+import ProfileCard from '../MyProfile/ProfileCard';
+import ApptCard from '../MyProfile/ApptCard';
+import ProfileModal from '../MyProfile/ProfileModal';
+import ApptModal from '../MyProfile/ApptModal';
+import api from '../api';
+
+export default function PatientProfileModal({ isOpen, onClose, patientId }) {
+  const [patient, setPatient] = useState(null);
+  const [appointments, setAppointments] = useState([]);
+  const [isProfileModalOpen, setProfileModalOpen] = useState(false);
+  const [isApptModalOpen, setApptModalOpen] = useState(false);
+  const [selectedAppointment, setSelectedAppointment] = useState(null);
+
+  useEffect(() => {
+    if (isOpen) {
+      const fetchPatientInfo = async () => {
+        try {
+          const response = await api.get(`/api/user/patients/${patientId}`, { withCredentials: true });
+          setPatient(response.data);
+
+          const appointmentsResponse = await api.get(`/api/user/patients/${patientId}/appointments`, { withCredentials: true });
+          setAppointments(appointmentsResponse.data);
+        } catch (error) {
+          console.error('Error fetching patient profile:', error.response ? error.response.data : error.message);
+        }
+      };
+
+      fetchPatientInfo();
+    }
+  }, [isOpen, patientId]);
+
+
+  //Edit Profile Command
+  const handleEditPatient = (formData) => {
+    // Implement the edit functionality
+  };
+
+  const handleDeletePatient = async () => {
```

Comment:
are we allowing physicians to delete patients? does this delete the patient from just the physician's list or completely from the application?

Reply:
similarly, I saw the past appointments in the video have a delete button; is deleting a previous appt an action physicians are allowed to do?

Reply:
Completely from the application. However, the user account will still exist.

Reply:
Okay, that makes sense. I'll implement that.